tests/core/test_sync_monitor.py

```diff
@@ -18,53 +18,48 @@
 
 
 @pytest.mark.trio
-async def test_monitors_idle(autojump_clock, running_backend, alice_core, alice):
-    autojump_clock.setup()
-
+async def test_monitors_idle(frozen_clock, running_backend, alice_core, alice):
     assert alice_core.are_monitors_idle()
 
     # Force wakeup of the sync monitor
     alice_core.event_bus.send(CoreEvent.FS_ENTRY_UPDATED, id=alice.user_manifest_id)
     assert not alice_core.are_monitors_idle()
-    with trio.fail_after(60):  # autojump, so not *really* 60s
+    await frozen_clock.sleep_with_autojump(60)
+    async with frozen_clock.real_clock_timeout():
         await alice_core.wait_idle_monitors()
     assert alice_core.are_monitors_idle()
 
 
 @pytest.mark.trio
-async def test_monitor_switch_offline(autojump_clock, running_backend, alice_core, alice):
-    autojump_clock.setup()
-
+async def test_monitor_switch_offline(frozen_clock, running_backend, alice_core, alice):
     assert alice_core.are_monitors_idle()
     assert alice_core.backend_status == BackendConnStatus.READY
 
     with alice_core.event_bus.listen() as spy:
         with running_backend.offline():
+            await frozen_clock.sleep_with_autojump(60)
             await spy.wait_with_timeout(
                 CoreEvent.BACKEND_CONNECTION_CHANGED,
                 {"status": BackendConnStatus.LOST, "status_exc": spy.ANY},
-                timeout=60,  # autojump, so not *really* 60s
             )
             await alice_core.wait_idle_monitors()
             assert alice_core.backend_status == BackendConnStatus.LOST
 
         # Switch backend online
 
+        await frozen_clock.sleep_with_autojump(60)
         await spy.wait_with_timeout(
             CoreEvent.BACKEND_CONNECTION_CHANGED,
             {"status": BackendConnStatus.READY, "status_exc": None},
-            timeout=60,  # autojump, so not *really* 60s
```

Comment:
Maybe `spy.wait_with_timeout` could use the frozen clock directly?
```python
spy.wait_with_timeout(..., autojump_timeout=60, side_effects_timeout=3)
```

Reply:
Those two clocks concept are complex enough, I'd rather keep them separated (that's part of the reasons I removed the possibility to customize the timeout in `wait_with_timeout`: otherwise this parameter really looks like the time elapsed with the trio clock)

So now everywhere in the tests we see something about timeout, it's related to side effects and real time clock, that's an easy rule of thumb ^^